### PR TITLE
- added luks resize support 

### PR DESCRIPTION
**Pull request by aschnell**
For https://trello.com/c/KDsLDxVQ/414-5-yast2-storage-ng-lvm-plain-devices-luks.

As usual, individual commits are consistent.

### CODE REVIEW

File: storage/Devices/EncryptionImpl.cc [hunk 1]
```diff
@@ -196,6 +196,52 @@ namespace storage
...
+
+	switch (resize_mode)
+	{
+	    case ResizeMode::SHRINK:
+		text = tenser(tense,
+			      // TRANSLATORS: displayed before action,
+			      // %1$s is replaced by partition name (e.g. /dev/sda1),
+			      // %2$s is replaced by old size (e.g. 2GiB),
+			      // %3$s is replaced by new size (e.g. 1GiB)
+			      _("Shrink encryption on %1$s from %2$s to %3$s"),
```
Comment (mvidner) [inline storage/Devices/EncryptionImpl.cc]: As I [said before](https://github.com/openSUSE/libstorage-ng/pull/142#r78337392), this does not make sense in English. Encryption cannot be shrunk. An encryption layer, or an encrypted device, can.

File: storage/Devices/LuksImpl.cc [hunk 2]
```diff
@@ -195,6 +207,22 @@ namespace storage
...
 
     void
+    Luks::Impl::do_resize(ResizeMode resize_mode) const
+    {
+	string cmd_line = CRYPTSETUPBIN " resize " + quote(get_dm_table_name());
+
+	if (resize_mode == ResizeMode::SHRINK)
+	    cmd_line += " --size " + to_string(get_size() / (512 * B));
+
+	cout << cmd_line << endl;
```
Comment (mvidner) [inline storage/Devices/LuksImpl.cc]: Forgotten debugging output?

File: storage/Devices/EncryptionImpl.cc (hunk 1 above)
Comment (aschnell) [inline storage/Devices/EncryptionImpl.cc]: Changed.

File: storage/Devices/LuksImpl.cc (hunk 2 above)
Comment (aschnell) [inline storage/Devices/LuksImpl.cc]: No, I'm keeping those outputs in all commit actions for the time being.

**Comment by aschnell** [comment]
Checks work fine for me, both locally and during build.

**Comment by mvidner** [comment]
OK, then I need to fix my setup. Thanks!

LGTM

**Pull request merged**